lib/internal/fs/dir.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const pathModule = require('path');
+const binding = internalBinding('fs');
+const dirBinding = internalBinding('fs_dir');
+const {
+  codes: {
+    ERR_MISSING_ARGS
+  }
+} = require('internal/errors');
+
+const { FSReqCallback } = binding;
+const internalUtil = require('internal/util');
+const {
+  getDirent,
+  getOptions,
+  getValidatedPath,
+  handleErrorFromBinding,
+  makeCallback
+} = require('internal/fs/utils');
+
+const kDirHandle = Symbol('Dir handle');
+const kDirPath = Symbol('Dir path');
+const kDirOptions = Symbol('Dir options');
+const kDirReadP = Symbol('Dir Awaitable Read');
+const kDirCloseP = Symbol('Dir Awaitable Close');
```

Comment:
Can you use 

```suggestion
const kDirCloseP = Symbol('kDirCloseP');
```

(or, imo even better,

```js
const kDirReadPromisified = Symbol('kDirReadPromisified');
```
)

and equivalents for the other symbols? It makes grepping the code for the Symbol easier and is what we’re doing for new code in general (.